pkg/nsx/services/inventory/builder.go

```diff
@@ -448,6 +416,67 @@ func (s *InventoryService) BuildService(service *corev1.Service) (retry bool) {
 	return
 }
 
+// determineServiceStatus determines the initial service status based on pod IDs, addresses, and selectors
+func (s *InventoryService) determineServiceStatus(podIDs []string, hasAddr, hasSelector bool, networkErrors *[]common.NetworkError, uniqueErrors map[string]bool) (string, string) {
+	status, netStatus := "", ""
+	if len(podIDs) > 0 {
+		status = InventoryStatusUp
+		netStatus = NetworkStatusHealthy
+	} else if hasAddr {
+		status = InventoryStatusUnknown
+		netStatus = NetworkStatusUnhealthy
+		errorMessage := "Service endpoint status is unknown"
+		uniqueErrors[errorMessage] = true
+		*networkErrors = append(*networkErrors, common.NetworkError{
+			ErrorMessage: errorMessage,
+		})
+	} else if hasSelector {
+		// Only mark as down if service has selector but no endpoints
+		status = InventoryStatusDown
+		netStatus = NetworkStatusUnhealthy
+		errorMessage := "Failed to get endpoints for Service"
+		uniqueErrors[errorMessage] = true
+		*networkErrors = append(*networkErrors, common.NetworkError{
+			ErrorMessage: errorMessage,
+		})
+	}
+	return status, netStatus
```

Comment:
what's if has ep, but no pidIDs, no selector?
it seesm return "", ""
the default value of status, netStatus  maybe be InventoryStatusDown ,  NetworkStatusUnhealthy?

Reply:
fixed